lxd/device/device_utils_network.go

```diff
@@ -917,3 +920,70 @@ func networkSRIOVSetupContainerVFNIC(hostName string, config map[string]string)
 
 	return nil
 }
+
+func isIPAvailable(ctx context.Context, address net.IP, parentInterface string) error {
+	if address.To4() != nil {
+		errs := make(chan error, 1)
+		go func() {
+			_, _, err := arping.PingOverIfaceByName(address, parentInterface)
+			if err == nil {
+				errs <- fmt.Errorf("ipv4.address %q is already in use", address.String())
+			}
+			errs <- nil
+		}()
+
+		select {
+		case err := <-errs:
+			return err
+		case <-ctx.Done():
+			return nil
+		}
```

Comment:
@tomponline @stgraber this is to get around `arping` which expects `arping.SetTimeout` to be called (setting a global value in the library). We could instead get a timeout to pass to `arping` via the `context.Deadline` and skip the extra goroutine, though this would mean the context wouldn't be cancellable.

Reply:
Yeah, that package's timeout handling is a bit problematic but should be fine, the function will timeout when we expect it to through the context and the goroutine will die some time later when the global timeout hits.

Reply:
Yep that's the best we can do here.